test/bats/apigateway/apigateway.bats

```diff
@@ -0,0 +1,195 @@
+
+    assert_output -p "\"name\": \"cli-test-gateway\""
+    assert_output -p "\"status\": \"PROVISIONING\""
+
+    echo "created apigateway gateway $gateway_id ($gateway_name)"
+    echo "$gateway_id" > /tmp/bats_test/gateway_id
+    echo "$gateway_name" > /tmp/bats_test/gateway_name
+}
+
+@test "List ApiGateway Gateways" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+
+    # List Gateway (JSON output)
+    run ionosctl apigateway gateway list -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"name\": \"cli-test-gateway\""
+
+    # List Gateway (Column output)
+    run ionosctl apigateway gateway list --cols name --no-headers
+    assert_success
+    assert_output -p "cli-test-gateway"
+}
+
+@test "Get ApiGateway Gateways" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+
+    # Get Gateway (JSON output)
+    run ionosctl apigateway gateway get --gateway-id "$gateway_id" -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"status\": \"AVAILABLE\""
+}
+
+@test "Create ApiGateway Route" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+    route_name="route"
+    run ionosctl apigateway route create --gateway-id "$gateway_id" --name "name_route" --paths "path_route" --methods "GET" --host "host.com" -o json 2> /dev/null
+    assert_success
+
+    route_id=$(echo "$output" | jq -r '.id')
+
+
+    # Verify specific fields
+    assert_output -p "\"name\": \"name_route\""
+    assert_output -p "\"path_route\""
+    assert_output -p "\"GET\""
+    assert_output -p "\"host\": \"host.com\""
+    assert_output -p "\"status\": \"PROVISIONING\""
+
+    echo "created apigateway route $route_id ($route_name)"
+    echo "$route_id" > /tmp/bats_test/route_id
+    echo "$route_name" > /tmp/bats_test/route_name
+}
+
+@test "List ApiGateway Routes" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+
+    # List Records (JSON output)
+    run ionosctl apigateway route list --gateway-id "$gateway_id" -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"name\": \"name_route\""
+
+    # List Records (Column output)
+    run ionosctl apigateway route list --gateway-id "$gateway_id" --cols name --no-headers
+    assert_success
+    assert_output "name_route"
+}
+
+
+@test "Get ApiGateway Route" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+    route_id=$(cat /tmp/bats_test/route_id)
+
+    # Get Route by ID
+    run ionosctl apigateway route get --gateway-id "$gateway_id" --route-id "$route_id" -o json 2> /dev/null
+    assert_success
+    assert_output -p "\"name\": \"name_route\""
+}
+
+@test "Gateway CustomDomains Operations" {
+    gateway_id=$(cat /tmp/bats_test/gateway_id)
+    certificate_id="bc4738cc-0a51-4f0b-ac20-ae65d930454e"
+
+    # Add CustomDomains
+    run ionosctl apigateway gateway customdomains add --gateway-id "$gateway_id" --name name.com --certificate-id "$certificate_id" -o json 2> /dev/null
+    assert_success
+    assert_output -p "name.com"
```

Comment:
again, check the key as well, not just the value. and please do it for all other checks in this test